### DISCUSSION TITLE
Store multi vectors with the same id

### DISCUSSION THREAD

**Discussion by sandraplato1234** [facebookresearch/faiss#1659] [Q&A]
Hi
I want for each image store for example 5 augmentations and all these vectors i want to index with the same id because they represent the same image. is that possible in faiss?
Thanks in advance

**Comment by Enet4** [comment]
To the best of my knowledge, vector IDs must be unique in an index. That would nevertheless be best implemented as an additional dictionary (e.g. a hash map), thus mapping each vector to some object.

**Reply by sandraplato1234** [reply]
Ok but when searching and asking for top_k it can return k from the same "image" and i need that will return distinct k images at all and not augmented images (that logically are the same image (vector representation)

**Reply by Enet4** [reply]
That would be related to [this FAQ](https://github.com/facebookresearch/faiss/wiki/FAQ#is-it-possible-to-dynamically-exclude-vectors-based-on-some-criterion). You would need to query a larger number of vectors and filter by the first occurrence of an image.
